BrainPortal/app/models/cluster_task.rb

```diff
@@ -1935,6 +1935,67 @@ def remove_cluster_workdir
     true
   end
 
+  # Save the directory created to run the job.
+  # The directory will be saved as a FileCollection
+  # only if the task have a results Data Provider.
+  def save_cluster_workdir(user_id="")
+    full_cluster_workdir = self.full_cluster_workdir
+    user                 = User.find(user_id)
+
+    # Some verification before saving the work directory
+    cb_error "Tried to save a task's work directory while in the wrong Rails app." unless
+      self.bourreau_id == CBRAIN::SelfRemoteResourceId
+
+    if full_cluster_workdir.blank?
+      self.addlog("Cannot save work directory: no task workdir was found")
+      return
+    end
+
+    data_provider_id = self.results_data_provider_id || user.meta[:pref_data_provider_id]
+    if data_provider_id.blank?
+      self.addlog("Cannot save work directory: the task should have a result Data Provider or user should have a default Data Provider")
+      return
+    end
+
+    if self.workdir_archived
+      self.addlog "Cannot save work directory: this task is archived"
+      return
+    end
+
+    if self.share_wd_tid
+      self.addlog "Cannot save work directory: this task doesn't have its own work directory"
+      return
+    end
+
+    # Save under a new FileCollection
+    userfile_name = self.tname_run_id
```

Comment:
Just using `tname_run_id` for the output name is too dangerous. I can imagine a task programmer choosing this for the name of the task's output, too. So your method here woudl crush the output.

I suggest renaming it:

```ruby
userfile_name = "TaskRawWorkdir-#{self.tname_run_id}"
```

Reply:
Also, maybe it woudl be worth it create a userfile model in cbrain-plugins-base:

``` ruby
TaskRawWorkdir < FileCollection
```